pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -80,8 +83,8 @@ public Consumer subscribe() throws PulsarClientException {
     }
 
     @Override
-    public CompletableFuture<Consumer> subscribeAsync() {
-        if (conf.getTopicNames().isEmpty() && conf.getTopicsPattern() == null) {
+    public CompletableFuture<Consumer<T>> subscribeAsync() {
+        if (conf.getTopicNames().isEmpty()) {
```

Comment:
did the `&& conf.getTopicsPattern() == null` got mangled in the merge?

Reply:
yeah, this became apparent when running tests